mlir/include/mlir/Dialect/Affine/Utils.h

```diff
@@ -309,6 +311,11 @@ DivModValue getDivMod(OpBuilder &b, Location loc, Value lhs, Value rhs);
 FailureOr<SmallVector<Value>> delinearizeIndex(OpBuilder &b, Location loc,
                                                Value linearIndex,
                                                ArrayRef<Value> basis);
+// Generate IR that extracts the linear index form a multi-index according to
+// a shape.
+OpFoldResult linearIndexFromShape(ArrayRef<OpFoldResult> multiIndex,
```

Comment:
I renamed it. `delinearizeIndex` above it is uses shape/basis as well. I changed the argument to basis for consistency.
There is another function `computeLinearIndex` in indexing utils that uses stride instead of basis.